src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -173,8 +182,15 @@ export default function OpeningPeriodForm({
     setValue,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     getValues,
+    watch,
   } = formMethods;
 
+  const clearResourceState = (): void => {
+    setValue(openingPeriodResourceStateKey, 'undefined');
+  };
+
+  const resourceStateValue = watch(openingPeriodResourceStateKey);
```

Comment:
Does watch work so that it creates a re-render of this component when the value associated with the openingPeriodResourceStateKey changes? That way then for example on line 327 can re-render itself? So with this watch we can avoid the need of applying useState in our components and can apply the watch instead, atleast in some cases?

Reply:
Yes, it is specially meant for that.